src/cli_util.rs

```diff
@@ -975,21 +979,82 @@ impl WorkspaceCommandHelper {
         }
     }
 
-    pub fn resolve_revset(&self, revision_str: &str) -> Result<Vec<Commit>, CommandError> {
-        let revset_expression = self.parse_revset(revision_str)?;
+    pub fn resolve_revset(
+        &self,
+        revision_str: &str,
+        ui: &mut Ui,
+    ) -> Result<Vec<Commit>, CommandError> {
+        let revset_expression = self.parse_revset(revision_str, Some(ui))?;
         let revset = self.evaluate_revset(revset_expression)?;
         Ok(revset.iter().commits(self.repo().store()).try_collect()?)
     }
 
     pub fn parse_revset(
         &self,
         revision_str: &str,
+        ui: Option<&mut Ui>,
     ) -> Result<Rc<RevsetExpression>, RevsetParseError> {
         let expression = revset::parse(
             revision_str,
             &self.revset_aliases_map,
             Some(&self.revset_context()),
         )?;
+        if let Some(ui) = ui {
```

Comment:
I would add a TODO here, e.g `TODO: remove when the legacy : operator is gone`

Reply:
I added a TODO in the .pest file instead. Once we remove the `:` operator from there, the generated `Rule` variants will be deleted. I'm sure we'll remember to remove the `is_legacy` fields on the `RevsetExpression` variants then. And then rustc will tells us that `ui` is unused.